plugins/module_utils/ds8000.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import json
+import traceback
+
+from ansible.module_utils.basic import missing_required_lib
+
+REQUESTS_IMP_ERR = None
+try:
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    REQUESTS_IMP_ERR = traceback.format_exc()
+    HAS_REQUESTS = False
+
+PYDS8K_IMP_ERR = None
+try:
+    from pyds8k.client.ds8k.v1.client import Client
+    HAS_PYDS8K = True
+except ImportError:
+    PYDS8K_IMP_ERR = traceback.format_exc()
+    HAS_PYDS8K = False
+
+DEFAULT_BASE_URL = '/api/v1'
+DEFAULT_POOLS_URL = '/pools'
+
+
+class PyDs8k(object):
```

Comment:
This is an abstract class.
We should name it accordingly and enforce it by changing its __meta__